### PR TITLE
Update `ButtonStyleButton.scaledPadding` documentation. Migrate callers in flutter/flutter

### LINKED ISSUE #138547: Migration from `textScaleFactor` to `TextScalar.scale` not ready for `ButtonStyleButton.scaledPadding`
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Call `ButtonStyleButton.scaledPadding` from a context where you don't have any font size available to provide for the non-linear scaling:
```dart
    ButtonStyleButton.scaledPadding(
      const EdgeInsets.symmetric(horizontal: 16),
      const EdgeInsets.symmetric(horizontal: 8),
      const EdgeInsets.symmetric(horizontal: 4),
      MediaQuery.maybeOf(context)?.textScaler.scale(??),
    );
```

### Expected results

A possibility to use `ButtonStyleButton.scaledPadding` without using the deprecated `textScaleFactor` variable.

### Actual results

Currently it doesn't seem possible to use  `ButtonStyleButton.scaledPadding` without using the deprecated `textScaleFactor` variable.

### Code sample

<details><summary>Code sample</summary>
Now:

```dart
    ButtonStyleButton.scaledPadding(
      const EdgeInsets.symmetric(horizontal: 16),
      const EdgeInsets.symmetric(horizontal: 8),
      const EdgeInsets.symmetric(horizontal: 4),
      MediaQuery.maybeOf(context)?.textScaler.scale(??),
    );
```

Previously:

```dart
    ButtonStyleButton.scaledPadding(
      const EdgeInsets.symmetric(horizontal: 16),
      const EdgeInsets.symmetric(horizontal: 8),
      const EdgeInsets.symmetric(horizontal: 4),
      MediaQuery.maybeOf(context)?.textScaleFactor,
    );
```

</details>

### Screenshots or Video

_No response_

### Logs

Other people with the same issue:
https://stackoverflow.com/questions/77494443/how-to-migrate-from-textscalefactor-to-textscalar-scale

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.16.0, on Ubuntu 23.10 6.5.0-10-generic, locale
    en_US.UTF-8)
    • Flutter version 3.16.0 on channel stable at /home/spydon/bin/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (20 hours ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2
```

</details>

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Fixes https://github.com/flutter/flutter/issues/138547. 

It makes more sense to me for the default padding values to use the scaled font sizes instead of the `textScaleFactor`.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/material/elevated_button.dart [hunk 1]
```diff
@@ -245,10 +245,13 @@ class ElevatedButton extends ButtonStyleButton {
...
   /// each state, and "others" means all other states.
   ///
-  /// The `textScaleFactor` is the value of
-  /// `MediaQuery.textScaleFactorOf(context)` and the names of the
-  /// EdgeInsets constructors and `EdgeInsetsGeometry.lerp` have been
-  /// abbreviated for readability.
+  /// {@template flutter.material.elevated_button.default_font_size}
+  /// The "default font size" below refers to the font size specified in the
+  /// [defaultStyleOf] method (or 14.0 if unspecified), scaled by the
+  /// `MediaQuery.textScalerOf(context).scale` method. And the names of the
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/elevated_button.dart:251]:
```suggestion
  /// `MediaQuery.textScalerOf(context).scale` method. The names of the
```

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/text_button_test.dart
Comment (piedcipher) [inline packages/flutter/test/material/text_button_test.dart:1]: why does expect uses `8` instead of `12`?

**Pull request merged**